### PR TITLE
Use GitHub Actions for building Docker image

### LINKED ISSUE #376: Docker builds - refactor
Proposed changes:
- Use GH Actions to build Docker image
- Switch base image to Ubuntu 20.04 and python 3.8
- Refactor Dockerfiles in order to manage cache better

### PR DESCRIPTION
**Pull request by tczekajlo**
**Proposed changes**:
- Use Ubuntu 20.04 as a base image
- Use Python 3.8
- Build wheels for python packages
- Resolves #376 

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

**Review by virtualroot — APPROVED** [review]

**Comment by tczekajlo** [comment]
@wochinge Can you 👀  if you have a minute and check if everything is OK python-wise? I wanna be sure that I didn't miss something that I'm not aware of 😄

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -83,9 +83,9 @@ jobs:
     name: Run Tests
     runs-on: ubuntu-latest
     strategy:
-      max-parallel: 2
+      max-parallel: 3
```
Comment (wochinge) [inline .github/workflows/continuous-integration.yml:86]: how about removing `max-parallel`?

File: Dockerfile [hunk 2]
```diff
@@ -1,29 +1,78 @@
...
+  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
+  && rm -rf /wheels \
+  && rm -rf /root/.cache/pip/*
 
-# copy everything from /opt
+# final image
+FROM base
+
+# copy needed files
+COPY ./poetry.lock /app/
```
Comment (wochinge) [inline Dockerfile:72]: why do we need `poetry.lock` here?

Comment (tczekajlo) [reply Dockerfile:72]: I saw that the old image includes this and I though it because of security scans for python packages, but it seems to be not a case

**Review by wochinge — APPROVED** [review]
🚀

**Pull request merged**